src/core/model.js

```diff
@@ -258,9 +259,11 @@ export default class Model extends Object3D {
   }
 
   setAttributes(attributes = {}) {
-    Object.assign(this.attributes, attributes);
-    this._createBuffersFromAttributeDescriptors(attributes);
-    this.setNeedsRedraw();
+    if (Object.keys(attributes).length > 0) {
```

Comment:
This is called with an empty `attributes` object by the default `Layer.updateState` implementation in
```
      model.setAttributes(changedAttributes);
```
When no attribute has changed.

Reply:
I wonder if there is a faster way than creating an array with all the keys. Something like
```
function isObjectEmpty(object) {
  let empty = true;
  for (const key in object) {
    empty = false;
    break;
  }
}
```
Not sure if this makes for better perf in practice, but since you are trying to push back on object creation...